numpy/lib/npyio.py

```diff
@@ -2243,7 +2244,11 @@ def encode_unicode_cols(row_tup):
         output = output.view(MaskedArray)
         output._mask = outputmask
     if unpack:
-        return output.squeeze().T
+        if names is not None:
+            # For structured arrays, return an array for each field.
+            return [output[field] for field in names]
```

Comment:
Should we squeeze on this branch too?

Reply:
I think in this case `output` is always one-dimensional. However, you're right because if the input is just a scalar, the return value should be zero-dimensional, even if the dtype is structured.

I moved the squeezing before the cases, and added a second condition so that the structured array branch requires `len(names) > 1`, which better matches the behavior of `loadtxt`.